pkg/git/git.go

```diff
@@ -244,3 +253,34 @@ func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
 	f.Close()
 	return true, nil
 }
+
+func validateGitAuth(logger *zap.SugaredLogger, url string) error {
+	homepath, err := homedir.Dir()
+	if err != nil {
+		logger.Errorf("Unexpected error: getting the user home directory: %v", err)
+		return err
+	}
+	euid := os.Geteuid()
+	sshCred := true
+	// Special case the root user/directory
+	if euid == 0 {
+		if _, err := os.Stat("/root/.ssh"); os.IsNotExist(err) {
+			sshCred = false
+		}
+	} else {
+		if _, err := os.Stat(homepath + "/.ssh"); os.IsNotExist(err) {
+			sshCred = false
+		}
+	}
```

Comment:
@sbwsg correct me if I am wrong but we *could* use `/tekton/creds` path in order to use a "always" working path ? (although this makes `git-init` only works inside a tekton managed container)

Reply:
The `ensureHomeEnv func` will create ssh symlink in `/tekton/home` so I change the stat path here.